examples/extra_element_integer/random_cluster.cpp

```diff
@@ -0,0 +1,148 @@
+    if (!needs_re_init){
+        equation_systems.init();
+    }
+    else{
+        equation_systems.reinit();
+    }
+}
+
+void FindCluster(libMesh::Mesh &mesh, libMesh::DofMap &dof_map,
+                 libMesh::DofMap &local_dof_map, LinearImplicitSystem &system,
+                 LinearImplicitSystem &local_system,
+                 const std::string &variable_name,
+                 libMesh::EquationSystems &equation_systems,
+                 const unsigned int index) {
+
+  const unsigned int variable_num = system.variable_number(variable_name);
+
+  for (const auto &elem : mesh.element_ptr_range()) {
+
+    std::vector<dof_id_type> global_dof_indices;
+    dof_map.dof_indices(elem, global_dof_indices, variable_num);
+
+    std::vector<double> solution_value(1);
+    system.solution->get(global_dof_indices, solution_value);
+    int element_solution = static_cast<int>(solution_value[0]);
+
+    bool belong_to_a_cluster = false;
+    for (unsigned int side = 0; side < elem->n_sides(); ++side) {
+      const Elem *neighbor = elem->neighbor_ptr(side);
+      if (neighbor) {
+        std::vector<dof_id_type> neighbor_dof_indices;
+        dof_map.dof_indices(neighbor, neighbor_dof_indices, variable_num);
+
+        std::vector<double> neighbor_solution_value(1);
+        system.solution->get(neighbor_dof_indices, neighbor_solution_value);
+        int neighbor_element_solution =
+            static_cast<int>(neighbor_solution_value[0]);
+
+        if (element_solution == neighbor_element_solution) {
+          belong_to_a_cluster = true;
+          break;
+        }
+      }
+    }
+
+    std::vector<dof_id_type> local_dof_indices;
+    local_dof_map.dof_indices(elem, local_dof_indices);
+    if (!belong_to_a_cluster) {
+      elem->set_extra_integer(index, 0);
+      local_system.solution->set(local_dof_indices[0], 0);
+    } else {
+      /*To do:
+       * maybe figure out another way to find out what should be the value of
+       * the extra_integer rather than just element solution
+       * */
+      elem->set_extra_integer(index, static_cast<int>(element_solution));
+      local_system.solution->set(local_dof_indices[0], element_solution);
+    }
+  }
+
+  local_system.solution->close();
+}
+
+void CloseSystems(libMesh::Mesh &mesh, LinearImplicitSystem &system,
+                  libMesh::EquationSystems &equation_systems,
+                  bool print_system_info = true) {
+  system.solution->close();
+  if (print_system_info) {
+    mesh.print_info();
+    equation_systems.print_info();
+  }
+}
+
+int main(int argc, char **argv) {
+  LibMeshInit init(argc, argv);
+
+  Mesh mesh(init.comm());
+  unsigned int nx = 40;
+  unsigned int ny = 40;
+  srand(time(0));
+
+  CreateMesh(mesh, nx, ny);
+
+  EquationSystems equation_systems(mesh);
+
+  //creating pesudo solution field
+  LinearImplicitSystem &system = equation_systems.add_system<LinearImplicitSystem>("random_solution_field");
+  DofMap &dof_map = system.get_dof_map();
```

Comment:
I don't think you need this `DofMap` in this scope - can you just create it when you need it in other functions?